lib/Auth.js

```diff
@@ -152,19 +192,38 @@ class Auth {
     //
     //
     //
-    generateAuthCode() {
+    generateAuthCode(username) {
+        this.removeExpiredAuthCode();
+
         let authCode = this.genRandomString();
 
-        this._authCode = {
-            authCode: authCode,
+        this._authCode.set(authCode, {
+            username: username,
             expiresAt: new Date(Date.now() + (10 * 60000)), // 10 minutes
-        };
+        });
 
         return authCode;
     }
     //
     //
     //
+    removeExpiredAuthCode() {
+        const now = Date.now();
+        let toDel = [];
+        for (let authCode of this._authCode.keys()) {
+            const authCodeInfo = this._authCode.get(authCode);
+            const expirationDate = new Date(authCodeInfo.expiresAt);
+            if (expirationDate < now) {
+                toDel.push(authCode);
+            }
+        }
+        for (let authCode of toDel) {
+            this._authCod.delete(authCode);
```

Comment:
`this._authCod` or `this._authCode`?

Reply:
Yes, sorry, this is a typo.
I should fix it.
Claudio